proxy/logging/LogAccessHttp.cc

```diff
@@ -71,7 +71,7 @@ LogAccessHttp::LogAccessHttp(HttpSM *sm)
     m_proxy_resp_content_type_str(nullptr),
     m_proxy_resp_content_type_len(0),
     m_cache_lookup_url_canon_str(nullptr),
-    m_cache_lookup_url_canon_len(0)
+    m_cache_lookup_url_canon_len(-1)
```

Comment:
```
diff --git a/proxy/logging/LogAccessHttp.cc b/proxy/logging/LogAccessHttp.cc
index 95a620e..b3b7490 100644
--- a/proxy/logging/LogAccessHttp.cc
+++ b/proxy/logging/LogAccessHttp.cc
@@ -57,9 +57,9 @@ LogAccessHttp::LogAccessHttp(HttpSM *sm)
     m_server_response(nullptr),
     m_cache_response(nullptr),
     m_client_req_url_str(nullptr),
-    m_client_req_url_len(0),
+    m_client_req_url_len(-1),
     m_client_req_url_canon_str(nullptr),
-    m_client_req_url_canon_len(0),
+    m_client_req_url_canon_len(-1),
     m_client_req_unmapped_url_canon_str(nullptr),
     m_client_req_unmapped_url_canon_len(-1),
     m_client_req_unmapped_url_path_str(nullptr),
@@ -67,9 +67,9 @@ LogAccessHttp::LogAccessHttp(HttpSM *sm)
     m_client_req_unmapped_url_host_str(nullptr),
     m_client_req_unmapped_url_host_len(-1),
     m_client_req_url_path_str(nullptr),
-    m_client_req_url_path_len(0),
+    m_client_req_url_path_len(-1),
     m_proxy_resp_content_type_str(nullptr),
-    m_proxy_resp_content_type_len(0),
+    m_proxy_resp_content_type_len(-1),
     m_cache_lookup_url_canon_str(nullptr),
     m_cache_lookup_url_canon_len(-1)
 {
```

But, I think I'd prefer to defer this until after we land this, to minimize last minute breakage of 7.1.0.